### DISCUSSION TITLE
ValueError: TypeError "'Decimal64' object is not iterable" when trying to return an array of dicts

### DISCUSSION THREAD

**Discussion by TrevorRichardson0480** [fastapi/fastapi#6266] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import grpc
from pygnmi.client import gNMIclient

app = FastAPI()

@app.post('/opr/state/cos/run/gnmi/')
def gnmi_data(input_gnmi: dict):
    subscribe = {
        'subscription': [
            {
                'path': input_gnmi['path'],
                'mode': 'sample',
                'sample_interval': 10000000000
            }
        ],
        'use_aliases': False,
        'mode': 'once',
        'encoding': 'proto'
    }

    with gNMIclient(target=(input_gnmi['ip'], input_gnmi['port']), username=input_gnmi['username'], password=input_gnmi['password'], insecure=True) as gc:
        telemetry_stream = gc.subscribe_once(subscribe=subscribe)

        returnThis = []

        for telemetry_entry in telemetry_stream:
            returnThis.append(telemetry_entry)
            if (str(telemetry_entry) == "{'sync_response': True}"): break

        return returnThis
    
    return ""
```

### Description

Trying to pull gNMI data from a network device. The actually gnmi stuff was tested and works but my issue comes up at the end of the example code. The variable 'telemetry_entry' is just a dict, 'returnThis' is an array of all the 'telemetry_entry's, so an array of dicts... 

I found that the error is trigger right after the loop on the line the 'return returnThis'. For some reason, jsonable_encoder thinks this is a Decimal64. The exact input I'm giving it follow this format:
`{"username":"", "password":"", "ip":"x.y.z.w", "port":"57400", "path":"/network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/"}`

Now this has worked fine for other paths like 
`network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/neighbor[neighbor-address=1.1.1.1]/afi-safis/afi-safi[afi-safi-name=IPV4_UNICAST]/state/prefixes/accepted`
which produce a smaller amount of data.

I'm wondering if there is something in the array of dicts produced when using this specific path that confuses jsonable_encoder
`/network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/`

I can fix this temporarily by going into line 144 in jsonable_encoder and replacing the `raise ValueError(errors)` with `return str(obj)`, and it works like normal, I can get the telemetry fine.

I can show you the data I'm trying to return, but it's a lot. Has something like this ever been seen before?

```Traceback (most recent call last):
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 137, in jsonable_encoder
    data = dict(obj)
TypeError: 'Decimal64' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 141, in jsonable_encoder
    data = vars(obj)
TypeError: vars() argument must have __dict__ attribute

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/trich/.local/lib/python3.7/site-packages/hypercorn/asyncio/task_group.py", line 21, in _handle
    await invoke_asgi(app, scope, receive, send)
  File "/home/trich/.local/lib/python3.7/site-packages/hypercorn/utils.py", line 247, in invoke_asgi
    await app(scope, receive, send)
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "/home/trich/.local/lib/python3.7/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/routing.py", line 244, in app
    is_coroutine=is_coroutine,
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/routing.py", line 149, in serialize_response
    return jsonable_encoder(response_content)
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 124, in jsonable_encoder
    sqlalchemy_safe=sqlalchemy_safe,
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 107, in jsonable_encoder
    sqlalchemy_safe=sqlalchemy_safe,
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 107, in jsonable_encoder
    sqlalchemy_safe=sqlalchemy_safe,
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 124, in jsonable_encoder
    sqlalchemy_safe=sqlalchemy_safe,
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 107, in jsonable_encoder
    sqlalchemy_safe=sqlalchemy_safe,
  File "/home/trich/.local/lib/python3.7/site-packages/fastapi/encoders.py", line 144, in jsonable_encoder
    raise ValueError(errors)
ValueError: [TypeError("'Decimal64' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
```

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.7.3

**Comment by iudeen** [comment]
Can you print the output of what you're trying to send as a response and add to your question? It'd be easier for us to replicate.

**Comment by TrevorRichardson0480** [comment]
> Can you print the output of what you're trying to send as a response and add to your question? It'd be easier for us to replicate.

This is output from one of the neighbors/peers

`{'update': {'update': [{'path': 'snmp-peer-index', 'val': 1}, {'path': 'state/peer-as', 'val': 65488}, {'path': 'state/local-as', 'val': 65488}, {'path': 'state/peer-type', 'val': 'INTERNAL'}, {'path': 'state/auth-password', 'val': 'configured'}, {'path': 'state/route-flap-damping', 'val': False}, {'path': 'state/description', 'val': 'Route reflector 1'}, {'path': 'state/session-state', 'val': 'CONNECT'}, {'path': 'state/last-established', 'val': 1633631105}, {'path': 'state/established-transitions', 'val': 0}, {'path': 'state/supported-capabilities'}, {'path': 'state/messages/sent/UPDATE', 'val': 0}, {'path': 'state/messages/sent/NOTIFICATION', 'val': 0}, {'path': 'state/messages/sent/last-notification-time', 'val': 0}, {'path': 'state/messages/sent/last-notification-error-subcode', 'val': 'UNSPECIFIC'}, {'path': 'state/messages/received/UPDATE', 'val': 0}, {'path': 'state/messages/received/NOTIFICATION', 'val': 0}, {'path': 'state/messages/received/last-notification-time', 'val': 0}, {'path': 'state/messages/received/last-notification-error-code', 'val': 'NONE'}, {'path': 'state/messages/received/last-notification-error-subcode', 'val': 'UNSPECIFIC'}, {'path': 'state/queues/input', 'val': 0}, {'path': 'state/queues/output', 'val': 0}, {'path': 'state/dynamically-configured', 'val': False}, {'path': 'state/session-status', 'val': 'RUNNING'}, {'path': 'state/session-admin-status', 'val': 'START'}, {'path': 'state/interface-error', 'val': False}, {'path': 'state/import-eval-pending', 'val': False}, {'path': 'state/import-eval', 'val': False}, {'path': 'state/peer-group', 'val': 'IBGP-ROUTE-REFLECTOR'}, {'path': 'state/neighbor-address', 'val': '1.1.1.1'}, {'path': 'state/enabled', 'val': True}, {'path': 'timers/state/connect-retry', 'val': precision: 2
}, {'path': 'timers/state/hold-time', 'val': digits: 9000
precision: 2
}, {'path': 'timers/state/keepalive-interval', 'val': digits: 3000
precision: 2
}, {'path': 'timers/state/minimum-advertisement-interval', 'val': precision: 2
}, {'path': 'timers/state/negotiated-hold-time', 'val': digits: 9000
precision: 2
}, {'path': 'transport/state/mtu-discovery', 'val': False}, {'path': 'transport/state/passive-mode', 'val': False}, {'path': 'transport/state/local-address', 'val': '140.222.240.133'}, {'path': 'transport/state/local-port', 'val': 0}, {'path': 'transport/state/remote-address', 'val': '1.1.1.1'}, {'path': 'transport/state/remote-port', 'val': 179}, {'path': 'error-handling/state/treat-as-withdraw', 'val': False}, {'path': 'error-handling/state/erroneous-update-messages', 'val': 0}, {'path': 'logging-options/state/log-neighbor-state-changes', 'val': True}, {'path': 'ebgp-multihop/state/enabled', 'val': True}, {'path': 'ebgp-multihop/state/multihop-ttl', 'val': 255}, {'path': 'route-reflector/state/route-reflector-cluster-id', 'val': 'zero-len'}, {'path': 'route-reflector/state/route-reflector-client', 'val': False}, {'path': 'as-path-options/state/allow-own-as', 'val': 0}, {'path': 'as-path-options/state/replace-peer-as', 'val': False}, {'path': 'as-path-options/state/disable-peer-as-filter', 'val': False}, {'path': 'use-multiple-paths/state/enabled', 'val': True}, {'path': 'use-multiple-paths/ebgp/state/allow-multiple-as', 'val': False}], 'timestamp': 1658933972612026611, 'prefix': 'network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/neighbor[neighbor-address=1.1.1.1]'}}{'update': {'update': [{'path': 'graceful-restart/state/enabled', 'val': True}, {'path': 'graceful-restart/state/received', 'val': False}, {'path': 'graceful-restart/state/advertised', 'val': False}, {'path': 'state/afi-safi-name', 'val': 'IPV4_UNICAST'}, {'path': 'state/enabled', 'val': True}, {'path': 'state/active', 'val': False}, {'path': 'state/prefixes/received', 'val': 0}, {'path': 'state/prefixes/sent', 'val': 0}, {'path': 'state/prefixes/installed', 'val': 0}, {'path': 'state/prefixes/accepted', 'val': 0}, {'path': 'state/prefix-limit-exceeded', 'val': False}, {'path': 'state/queues/input', 'val': 0}, {'path': 'state/queues/output', 'val': 0}, {'path': 'add-paths/state/receive', 'val': True}, {'path': 'add-paths/state/send', 'val': False}, {'path': 'add-paths/state/eligible-prefix-policy', 'val': ''}, {'path': 'ipv4-unicast/prefix-limit/state/max-prefixes', 'val': 0}, {'path': 'ipv4-unicast/prefix-limit/state/prevent-teardown', 'val': True}, {'path': 'ipv4-unicast/prefix-limit/state/shutdown-threshold-pct', 'val': 0}, {'path': 'ipv4-unicast/prefix-limit/state/restart-timer', 'val': precision: 2
}], 'timestamp': 1658933972616476378, 'prefix': 'network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/neighbor[neighbor-address=1.1.1.1]/afi-safis/afi-safi[afi-safi-name=IPV4_UNICAST]'}}{'update': {'update': [{'path': 'graceful-restart/state/enabled', 'val': True}, {'path': 'graceful-restart/state/received', 'val': False}, {'path': 'graceful-restart/state/advertised', 'val': False}, {'path': 'state/afi-safi-name', 'val': 'IPV6_LABELED_UNICAST'}, {'path': 'state/enabled', 'val': True}, {'path': 'state/active', 'val': False}, {'path': 'state/prefixes/received', 'val': 0}, {'path': 'state/prefixes/sent', 'val': 0}, {'path': 'state/prefixes/installed', 'val': 0}, {'path': 'state/prefixes/accepted', 'val': 0}, {'path': 'state/prefix-limit-exceeded', 'val': False}, {'path': 'state/queues/input', 'val': 0}, {'path': 'state/queues/output', 'val': 0}, {'path': 'add-paths/state/receive', 'val': False}, {'path': 'add-paths/state/send', 'val': False}, {'path': 'add-paths/state/eligible-prefix-policy', 'val': ''}, {'path': 'ipv6-labeled-unicast/prefix-limit/state/max-prefixes', 'val': 0}, {'path': 'ipv6-labeled-unicast/prefix-limit/state/prevent-teardown', 'val': True}, {'path': 'ipv6-labeled-unicast/prefix-limit/state/shutdown-threshold-pct', 'val': 0}, {'path': 'ipv6-labeled-unicast/prefix-limit/state/restart-timer', 'val': precision: 2
}], 'timestamp': 1658933972618497944, 'prefix': 'network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/neighbor[neighbor-address=1.1.1.1]/afi-safis/afi-safi[afi-safi-name=IPV6_LABELED_UNICAST]'}}{'update': {'update': [{'path': 'graceful-restart/state/enabled', 'val': True}, {'path': 'graceful-restart/state/received', 'val': False}, {'path': 'graceful-restart/state/advertised', 'val': False}, {'path': 'state/afi-safi-name', 'val': 'IPV4_FLOW'}, {'path': 'state/enabled', 'val': True}, {'path': 'state/active', 'val': False}, {'path': 'state/prefixes/received', 'val': 0}, {'path': 'state/prefixes/sent', 'val': 0}, {'path': 'state/prefixes/installed', 'val': 0}, {'path': 'state/prefixes/accepted', 'val': 0}, {'path': 'state/prefix-limit-exceeded', 'val': False}, {'path': 'state/queues/input', 'val': 0}, {'path': 'state/queues/output', 'val': 0}, {'path': 'add-paths/state/receive', 'val': False}, {'path': 'add-paths/state/send', 'val': False}, {'path': 'add-paths/state/eligible-prefix-policy', 'val': ''}], 'timestamp': 1658933972621030884, 'prefix': 'network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/neighbor[neighbor-address=1.1.1.1]/afi-safis/afi-safi[afi-safi-name=IPV4_FLOW]'}}{'update': {'update': [{'path': 'graceful-restart/state/enabled', 'val': False}, {'path': 'graceful-restart/state/restart-time', 'val': 120}, {'path': 'graceful-restart/state/stale-routes-time', 'val': digits: 30000
precision: 2
}, {'path': 'graceful-restart/state/helper-only', 'val': True}, {'path': 'graceful-restart/state/peer-restarting', 'val': False}, {'path': 'graceful-restart/state/local-restarting', 'val': False}, {'path': 'graceful-restart/state/mode', 'val': 'HELPER_ONLY'}, {'path': 'apply-policy/state/import-policy'}, {'path': 'apply-policy/state/export-policy'}], 'timestamp': 1658933972621522719, 'prefix': 'network-instances/network-instance[instance-name=master]/protocols/protocol/bgp/neighbors/neighbor[neighbor-address=1.1.1.1]'}}`

**Comment by TrevorRichardson0480** [comment]
there is a section from the gnmi response that looks like this:

`{
	'path': 'timers/state/hold-time',
	'val': digits: 9000
	precision: 2
}`

wondering if that could be causing it

**Comment by iudeen** [comment]
Yes, its not a valid JSON...

**Comment by TrevorRichardson0480** [comment, marked as answer]
Ok. It seems like gnmi is actually returning **proto** (not json) dictionaries which is probably causing this. Interesting though, as I was looking for a way to maybe convert proto to json, it suddenly started working... Not sure what's making the issue intermittent like this. 

Looking at swagger ui, this is how it's handling that "digits: 9000 precision: 2" part
![image](https://user-images.githubusercontent.com/60578933/181805893-7b1f5569-8f14-4738-9f30-802067bad547.png)

**Comment by iudeen** [comment]
It's not swagger, rather that's how your Proto is being converted to JSON. Any JSON viewer would display it this way.

**Comment by TrevorRichardson0480** [comment]
I'm aware of that. The way the response formatted is fine for how I'm using. Just confused as to why that error disappeared and if it will come back.
